src/neo/Network/P2P/Payloads/GetBlockByIndexPayload.cs

```diff
@@ -4,16 +4,17 @@
 
 namespace Neo.Network.P2P.Payloads
 {
-    public class GetBlockByIndexPayload : ISerializable
+    public class GetBlockDataPayload : ISerializable
```

Comment:
I think we don't need to change anything in the P2P protocol to implement header-first sync. We can get blocks by indexes with `GetBlockByIndex`, we can get blocks by hashes with `GetData` and we can get headers with `GetHeaders`. So protocol-wise we already have everything we need and we better not change it. Different node implementations can have different strategies for their synchronization, but the protocol is there already for any of them.